src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/CommentUtils.java

```diff
@@ -401,6 +503,59 @@ private boolean hasParamForComponent(TypeElement elem, Name component) {
         }
     }
 
+    /**
+     * Returns a list containing the string for a given key in the doclet's resources,
+     * formatted with given arguments.
+     *
+     * @param key the key for the desired string
+     * @param o0  string or tree argument to be formatted into the result
+     * @return a content tree for the text
+     */
+    public List<? extends DocTree> getComment(String key, Object o0) {
+        return getComment(key, o0, null, null);
+    }
+
+    /**
+     * Returns a list containing the string for a given key in the doclet's resources,
+     * formatted with given arguments.
+     *
+     * @param key the key for the desired string
+     * @param o0  string or tree argument to be formatted into the result
+     * @param o1  string or tree argument to be formatted into the result
+     * @param o2  string or tree argument to be formatted into the result
+     * @return a content tree for the text
+     */
+    public List<? extends DocTree> getComment(String key, Object o0, Object o1, Object o2) {
```

Comment:
What is the purpose of this rather complex method to format a text resource? It seems like it is arguments `o1` and `o2` are never used. Couldn't this be implemented simply using `MessageFormat` as it used to be?

Reply:
This is the `DocTree` equivalent of similar code in `Content`, to format structured objects into a format string. Previously, using `MessageFormat`, the code used "regular" text to inject the name of the property. Now, using this new code, we can inject the equivalent of `{@code name}`.

Yes, it is currently a bit more general than strictly required, in that currently we only require a single value to be injected. I guess I was just following the pattern of supporting future use.

Reply:
> What is the purpose of this rather complex method to format a text resource? It seems like it is arguments `o1` and `o2` are never used. Couldn't this be implemented simply using `MessageFormat` as it used to be?

This is the `DocTree` equivalent of similar code in `Content`, to format structured objects into a format string. Previously, using `MessageFormat`, the code used "regular" text to inject the name of the property. Now, using this new code, we can inject the equivalent of `{@code name}`.

Yes, it is currently a bit more general than strictly required, in that currently we only require a single value to be injected. I guess I was just following the pattern of supporting future use.

Reply:
Thanks for the explanation. Sounds reasonable.